server/handlers/authentication.go

```diff
@@ -84,11 +87,11 @@ func (a *authenticationHandlerImpl) postLoginHandler(w http.ResponseWriter, r *h
 		Value: token,
 		Path:  "/",
 	})
-
 	http.Redirect(w, r, next, http.StatusSeeOther)
 }
 
 func (a *authenticationHandlerImpl) parseLoginWithErr(w http.ResponseWriter, r *http.Request, err error) {
+	w.WriteHeader(http.StatusUnauthorized)
 	a.parseHTML(w,
```

Comment:
add we add params in parseHTML to receive http status code?

Reply:
I tried passing the statusCode into the `parseHTML` function, and using `w`, I tried to write it there but for some reason something is overriding the status code there and it is always coming out as 200.

Reply:
https://stackoverflow.com/questions/43020241/how-do-you-execute-a-template-with-500-status-code-in-go

This might be the solution.

Reply:
Just write the header above the call to "Execute" on Line 2 of html.go in server > resource_server > utils > html.go

```
w.WriteHeader(http.StatusInternalServerError)
t.Execute(w, page)
```